### ISSUE TITLE
Explain how ping & "session expired" popup works

### ISSUE THREAD

**Issue by warpech** [PuppetJs/PuppetJs#77]
On PuppetJs wiki, there is a document about how does [establishing the connection] work (https://github.com/PuppetJs/PuppetJs/wiki/Server-communication) and how does the [synchronization/OT](https://github.com/PuppetJs/PuppetJs/wiki/Server-communication:-advanced-synchronization) work.

There is no document about how the reconnection works and what event/callback is available to inform about the expiry of the session.

Basically, I mean documenting what was done in https://github.com/Starcounter/Starcounter/issues/3084

@miyconst can you do that?

This was brought to my attention by @mtornwall

**Comment by miyconst** [comment]
@warpech we do not have re-connection functionality. The only thing we have is backing up to http if websocket connection fails.

But if connection of the websocket drops after a patch was sent, then this session is no longer usable.

**Comment by miyconst** [comment]
Ping.

**Comment by warpech** [comment]
> @warpech we do not have re-connection functionality. The only thing we have is backing up to http if websocket connection fails.

I meant what was done in https://github.com/Starcounter/Starcounter/issues/3084. Since there is no reconnection, the only thing that was done is ping and the "session expired" popup. Can you explain on the wiki page how does the ping work and when is the "session expired" popup triggered?

> But if connection of the websocket drops after a patch was sent, then this session is no longer usable.

Isn't it? As far as I know, the session is still usable for the next 20 minutes (or whatever is the current setting) after last patch. Patches can still be sent using HTTP PATCH.

**Comment by miyconst** [comment]
Okay, this I can explain.

But:

>... when is the "session expired" popup triggered?

There is no such popup, the popup says: "You have been disconnected from server. Please reload page." and looks like this:

![image](https://cloud.githubusercontent.com/assets/6435556/12445310/e1a0e984-bf63-11e5-91f3-f5bd9f6206ec.png)

>Isn't it? As far as I know, the session is still usable for the next 20 minutes ...

The sessions lives on the server for the next 20 minutes, but we do not have any implementation to restore queue sync, once a patch was sent and was not received by one of the sides.

**Comment by miyconst** [comment]
@warpech do you think this is enough:

https://github.com/PuppetJs/PuppetJs/wiki/Keep-server-session-alive

**Comment by miyconst** [comment]
Closing this issue as done.
